### PR TITLE
Problem: ibc incentivization is not integrated

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #642
Solution:
- [x] integration ibc fee middleware
- [x] pass exiting integration test
- [x] add ibc incentivized integration test

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -229,6 +233,7 @@ func GenModuleBasics(experimental bool) module.BasicManager {
 		vesting.AppModuleBasic{},
 		ica.AppModuleBasic{},
 		icactlmodule.AppModuleBasic{},
+    ibcfee.AppModuleBasic{},
```
Comment (yihuang) [inline app/app.go:236]:
```suggestion
		ibcfee.AppModuleBasic{},
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 - [cronos#600](https://github.com/crypto-org-chain/cronos/pull/600) Implement bidirectional token mapping.
 - [cronos#611](https://github.com/crypto-org-chain/cronos/pull/611) Fix mistake on acknowledgement error in ibc middleware.
 - [cronos#627](https://github.com/crypto-org-chain/cronos/pull/627) Upgrade gravity bridge module with security enhancements
+- [cronos#627](https://github.com/crypto-org-chain/cronos/pull/627) Integrate ibc fee middleware.
```
Comment (yihuang) [inline CHANGELOG.md:11]:
```suggestion
- [cronos#647](https://github.com/crypto-org-chain/cronos/pull/647) Integrate ibc fee middleware.
```

File: app/app.go [hunk 3]
```diff
@@ -416,6 +423,14 @@ func New(
...
+	// IBC Fee Module keeper
+	app.IBCFeeKeeper = ibcfeekeeper.NewKeeper(
+		appCodec, keys[ibcfeetypes.StoreKey], app.GetSubspace(ibcfeetypes.ModuleName),
+		app.IBCKeeper.ChannelKeeper, // more middlewares can be added in future
+		app.IBCKeeper.ChannelKeeper,
+		&app.IBCKeeper.PortKeeper, app.AccountKeeper, app.BankKeeper,
+	)
+
 	// Create Transfer Keepers
 	app.TransferKeeper = ibctransferkeeper.NewKeeper(
```
Comment (devashishdxt) [inline app/app.go:435]: Fee middleware should be added here. Similar to this: https://github.com/crypto-org-chain/chain-main/blob/fc05db0f4a26956181b968f02d761625929f5089/app/app.go#L436

File: app/app.go [hunk 4]
```diff
@@ -494,8 +509,13 @@ func New(
...
+	feeModule := ibcfee.NewAppModule(app.IBCFeeKeeper)
+
+	var transferStack porttypes.IBCModule
+	transferStack = transfer.NewIBCModule(app.TransferKeeper)
+	transferStack = middleware.NewIBCConversionModule(transferStack, app.CronosKeeper)
+	transferStack = ibcfee.NewIBCMiddleware(transferStack, app.IBCFeeKeeper)
+
 	app.ICAControllerKeeper = icacontrollerkeeper.NewKeeper(
 		appCodec, keys[icacontrollertypes.StoreKey], app.GetSubspace(icacontrollertypes.SubModuleName),
 		app.IBCKeeper.ChannelKeeper, // may be replaced with middleware such as ics29 fee
```
Comment (devashishdxt) [inline app/app.go:521]: Same as above

**Review by devashishdxt — COMMENTED** [review]
Integration test seems fine. Just a small comment on app wiring.

File: app/app.go (hunk 3 above)
Comment (yihuang) [reply app/app.go:435]: fixed, thanks

File: app/app.go (hunk 4 above)
Comment (yihuang) [reply app/app.go]: fixed

**Review by devashishdxt — APPROVED** [review]

**Review by devashishdxt — COMMENTED** [review]
Fee middleware is supposed to be backwards compatible. It'll be better if we also write some tests with `incentivized=False`

**Comment by yihuang** [comment]
> Fee middleware is supposed to be backwards compatible. It'll be better if we also write some tests with `incentivized=False`. I ran into some problems with non-fee enabled channels on `chain-main`. The issue was faulty app wiring.

I parameterized the ibc fixture, to run test cases on both cases.

**Review by mmsqe — APPROVED** [review]

**Pull request merged**